src/App.tsx

```diff
@@ -1,61 +1,118 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import { useState } from 'react';
+import { todosWithUser } from './services/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { newIdTodo } from './services/newId';
+import { filterInput } from './services/filterInput';
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUser);
+  const [newTodo, setNewTodo] = useState<Todo>({
+    id: 0,
+    title: '',
+    completed: false,
+    user: undefined,
+    userId: 0,
+  });
+  const [count, setCount] = useState(0);
+  const [validTitle, setValidTitle] = useState(false);
+  const [validSelect, setValidSelect] = useState(false);
+
+  const handleChangeNewTodo = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    const filterValue = filterInput(value);
+
+    setNewTodo(prev => ({
+      ...prev,
+      [name]: filterValue,
+      id: newIdTodo(todos),
+    }));
+    setValidTitle(filterValue === '');
+  };
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = +event.target.value;
+
+    setNewTodo(prev => ({
+      ...prev,
+      user: usersFromServer.find(user => user.id === value),
+      userId: value,
+    }));
+    setValidSelect(value === 0);
+  };
+
+  const reset = () => {
+    setNewTodo({
+      id: 0,
+      title: '',
+      completed: false,
+      user: undefined,
+      userId: 0,
+    });
+    setValidTitle(false);
+    setValidSelect(false);
+  };
+
+  const onSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setValidTitle(newTodo.title === '');
+    setValidSelect(newTodo.userId === 0);
+
+    if (newTodo.title === '' || newTodo.userId === 0) {
```

Comment:
i can create the todo only with spaces, let's fix that
<img width="213" alt="image" src="https://github.com/mate-academy/react_add-todo-form/assets/70220667/1258abd2-d9d5-401b-90de-cbcd93a5f59e">
```suggestion
    if (newTodo.title.trim() === '' || newTodo.userId === 0) {
```